gapis/api/transform2/transform_chain.go

```diff
@@ -26,30 +26,48 @@ import (
 // TransformChain is responsible for running all the transforms on
 // all the commands produced by command enerator
 type TransformChain struct {
-	transforms       []Transform
-	out              Writer
-	generator        commandGenerator.CommandGenerator
-	hasBegun         bool
-	hasEnded         bool
-	currentCommandID api.CmdID
+	transforms            []Transform
+	out                   Writer
+	generator             commandGenerator.CommandGenerator
+	hasBegun              bool
+	hasEnded              bool
+	currentCommandID      api.CmdID
+	currentTransformIndex int
+	mutator               StateMutator
 }
 
 // CreateTransformChain creates a transform chain that will run
 // the required transforms on every command coming from commandGenerator
-func CreateTransformChain(generator commandGenerator.CommandGenerator, transforms []Transform, out Writer) *TransformChain {
-	return &TransformChain{
+func CreateTransformChain(ctx context.Context, generator commandGenerator.CommandGenerator, transforms []Transform, out Writer) *TransformChain {
+	chain := TransformChain{
 		generator:        generator,
 		transforms:       transforms,
 		out:              out,
 		hasBegun:         false,
 		hasEnded:         false,
 		currentCommandID: 0,
+		mutator:          nil,
 	}
+
+	chain.mutator = func(id api.CmdID, cmds []api.Cmd) error {
+		return chain.stateMutator(ctx, id, cmds)
+	}
+
+	for _, t := range chain.transforms {
+		if t.RequiresAccurateState() {
+			// Melih TODO: Temporary check until we implement accurate state
+			panic("Implement accurate state")
+		}
+
+		if t.RequiresInnerStateMutation() {
+			t.SetInnerStateMutationFunction(chain.mutator)
+		}
+	}
+
+	return &chain
 }
 
 func (chain *TransformChain) beginChain(ctx context.Context) error {
-	chain.handleInitialState(chain.out.State())
```

Comment:
Why does this get deleted? Seems unrelated to the CL

Reply:
I think we had talked about it. This is not in use therefore I removed it.

Reply:
I've commented below. I believe we should keep it in as a stub. We will need to implement this functionality later for transforms that NeedAccurateState()